config/config-zk/src/main/java/com/alipay/sofa/rpc/dynamic/zk/ZookeeperDynamicConfigManager.java

```diff
@@ -0,0 +1,112 @@
+import com.alipay.sofa.rpc.dynamic.DynamicHelper;
+import com.alipay.sofa.rpc.ext.Extension;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+
+/**
+ * @author Narziss
+ * @version ZookeeperDynamicConfigManager.java, v 0.1 2024年07月20日 09:23 Narziss
+ */
+
+@Extension(value = "zookeeper", override = true)
+public class ZookeeperDynamicConfigManager extends DynamicConfigManager {
+
+    private CuratorFramework zkClient;
+    private static final String ADDRESS = "127.0.0.1:2181";
+    private static final String DEFAULT_NAMESPACE = "sofa-rpc";
+    private static final String CONFIG_NODE = "/config";
+    private static final String Default_GROUP = "/application";
+
+    protected ZookeeperDynamicConfigManager(String appName) {
+        super(appName);
+        zkClient = CuratorFrameworkFactory.builder()
+                .connectString(ADDRESS)
+                .retryPolicy(new ExponentialBackoffRetry(1000, 3))
+                .namespace(DEFAULT_NAMESPACE)
+                .build();
+        zkClient.start();
+    }
+
+    @Override
+    public void initServiceConfiguration(String service) {
+        //TODO not now
+    }
+
+    @Override
+    public String getProviderServiceProperty(String service, String key) {
+        try {
+            String path = getPrefixPath() + DynamicConfigKeyHelper.buildProviderServiceProKey(service, key);
+            byte[] bytes = zkClient.getData().forPath(path);
+            return bytes != null ? new String(bytes) : DynamicHelper.DEFAULT_DYNAMIC_VALUE;
+        } catch (Exception e) {
+            return DynamicHelper.DEFAULT_DYNAMIC_VALUE;
+        }
+    }
+
+    @Override
+    public String getConsumerServiceProperty(String service, String key) {
+        try {
+            String path = getPrefixPath() + DynamicConfigKeyHelper.buildConsumerServiceProKey(service, key);
+            byte[] bytes = zkClient.getData().forPath(path);
+            return bytes != null ? new String(bytes) : DynamicHelper.DEFAULT_DYNAMIC_VALUE;
+        } catch (Exception e) {
+            return DynamicHelper.DEFAULT_DYNAMIC_VALUE;
+        }
+    }
+
+    @Override
+    public String getProviderMethodProperty(String service, String method, String key) {
+        try {
+            String path = getPrefixPath() + DynamicConfigKeyHelper.buildProviderMethodProKey(service, method, key);
+            byte[] bytes = zkClient.getData().forPath(path);
+            return bytes != null ? new String(bytes) : DynamicHelper.DEFAULT_DYNAMIC_VALUE;
+        } catch (Exception e) {
+            return DynamicHelper.DEFAULT_DYNAMIC_VALUE;
+        }
+    }
+
+    @Override
+    public String getConsumerMethodProperty(String service, String method, String key) {
+        try {
+            String path = getPrefixPath() + DynamicConfigKeyHelper.buildConsumerMethodProKey(service, method, key);
```

Comment:
This `zkClient.getData().forPath`  method does not have cached. Here should add a cache in conjunction with the listener mentioned above.